ddev/src/ddev/cli/validate/metadata.py

```diff
@@ -0,0 +1,294 @@
+        reader._fieldnames = reader.fieldnames
+
+        for line_no, row in enumerate(reader, 2):
+            yield line_no, row
+
+
+def normalize_metric_name(metric_name):
+    """Copy pasted from the backend normalization code.
+    Extracted from dogweb/datalayer/metrics/query/metadata.py:normalize_metric_name
+    Metrics in metadata.csv need to be formatted this way otherwise, syncing metadata will fail.
+    Function not exported as a util, as this is different than AgentCheck.normalize. This function just makes sure
+    that whatever is in the metadata.csv is understandable by the backend.
+    """
+    from ddev.cli.validate import metadata_utils
+
+    if not isinstance(metric_name, str):
+        metric_name = str(metric_name)
+    metric_name = metadata_utils.METRIC_REPLACEMENT.sub("_", metric_name)
+    return metadata_utils.METRIC_DOTUNDERSCORE_CLEANUP.sub(".", metric_name).strip("_")
+
+
+def check_duplicate_values(current_check, line, row, header_name, duplicates, fail=None):
+    """Check if the given column value has been seen before.
+    Output a warning and return True if so.
+    """
+    if row[header_name] and row[header_name] not in duplicates:
+        duplicates.add(row[header_name])
+    elif row[header_name] != '':
+        message = f"{current_check}:{line} `{row[header_name]}` is a duplicate {header_name}"
+        if fail:
+            Application.display_error(message)
+            return True
+        else:
+            Application.display_warning(message)
+    return False
+
+
+@click.command(short_help='Validate `metadata.csv` files')
+@click.argument('integrations', nargs=-1)
+@click.option(
+    '--check-duplicates', is_flag=True, help='Output warnings if there are duplicate short names and descriptions'
+)
+@click.option('--show-warnings', '-w', is_flag=True, help='Show warnings in addition to failures')
+@click.pass_obj
+def metadata(app: Application, integrations: tuple[str, ...], check_duplicates: bool, show_warnings: bool):
+    """
+    Validate `metadata.csv` files
+
+    If `check` is specified, only the check will be validated, if check value is
+    'changed' will only apply to changed checks, an 'all' or empty `check` value
+    will validate all README files.
+    """
+    from ddev.cli.validate import metadata_utils
+
+    validation_tracker = app.create_validation_tracker('Metrics validation')
+
+    excluded = set(app.repo.config.get('/overrides/validate/metrics/exclude', []))
+    for current_check in app.repo.integrations.iter(integrations):
+        if current_check.name in excluded or not current_check.has_metrics:
+            continue
+
+        errors = False
+        error_message = ""
+
+        display_queue = []
+        if current_check.name.startswith('datadog_checks_'):
+            continue
+
+        try:
+            metric_prefix = current_check.manifest.get("/assets/integration/metrics/prefix", "").rstrip()
+        except KeyError:
+            metric_prefix = None
+
```

Comment:
This would be impossible to fail now I think

```suggestion
        metric_prefix = current_check.manifest.get("/assets/integration/metrics/prefix", "")
```